components/cli/vendor/github.com/docker/docker/client/client.go

```diff
@@ -260,8 +261,13 @@ func (cli *Client) NegotiateAPIVersionPing(p types.Ping) {
 		p.APIVersion = "1.24"
 	}
 
-	// if server version is lower than the current cli, downgrade
-	if versions.LessThan(p.APIVersion, cli.ClientVersion()) {
+	// if the client is not initialized with a version, start with the latest supported version
```

Comment:
This does not look correct to me, so I would recommend removing to be safe. First, in the `cli` use case always sets the version, so the new condition doesn't apply. The new `versions.LessThan(p.APIVersion, api.DefaultVersion)` does not look correct to me. If user sets a manual low version it should not be made bigger if daemon supports something that is is lower than default and higher than the manual override. The cli does not seem to set the `client.manualOverride` as well. cc @thaJeztah